build-container.ps1

```diff
@@ -11,10 +11,18 @@ $BaseTable = @{
     2004 = "mcr.microsoft.com/dotnet/framework/runtime:4.8-windowsservercore-2004";
     ## 20H2 reports itself as 2009 in the registry
     2009 = "mcr.microsoft.com/dotnet/framework/runtime:4.8-windowsservercore-20H2"
+    2022 = "mcr.microsoft.com/dotnet/framework/runtime:4.8-windowsservercore-ltsc2022"
 }
 
 $kernelver = [int](get-itemproperty -path "hklm:software\microsoft\windows nt\currentversion" -name releaseid).releaseid
-Write-Host -ForegroundColor Green "Detected kernel version $kernelver, using base image $($BaseTable[$kernelver])"
+$productname = (get-itemproperty -path "hklm:software\microsoft\windows nt\currentversion" -n productname).productname
+Write-Host -ForegroundColor Green "Detected kernel version $kernelver and product name $productname"
+// Windows Server 2022 still reports 2009 as releaseid
```

Comment:
```suggestion
# Windows Server 2022 still reports 2009 as releaseid
```

Reply:
These files were missing `$ErrorActionPreference = "Stop"` so this didn't make the script fail. I'll add it.